06_puppy/n0npax/memStore.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"errors"
+)
+
+func NewMemStore() MemStore {
+	return MemStore{}
+}
+
+func (m MemStore) CreatePuppy(puppy *Puppy) int {
+	id := len(m)
+	m[id] = puppy
+	return id
+}
+
+func (m MemStore) ReadPuppy(id int) (*Puppy, error) {
+	puppy, ok := m[id]
+	if !ok {
+		return nil, errors.New("doesn't exists")
+	}
+	return puppy, nil
+}
+
+func (m MemStore) UpdatePuppy(id int, puppy *Puppy) error {
+	m[id] = puppy
```

Comment:
Can you include an error check here to see if the puppy exists? Since we are returning an error, it suggests we should be doing this kind of check.
```go
if _, ok := m[id]; !ok {
    return errors.New(fmt.Sprintf("Puppy with ID: %d does not exist", id))
}
m[id] = puppy
return nil
```

Reply:
fixed. Thanks for that!